### PR TITLE
Fix error message in float8 FSDP utils

### LINKED ISSUE #1108: Assertion information on precompute_float8_dynamic_scale_for_fsdp is misleading
https://github.com/pytorch/ao/blob/main/torchao/float8/fsdp_utils.py#L44-L48

Should it be `raise NotImplementedError("Only supports dynamic scaling")`?

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Fixes #1108 

Error message should say `"Only supports dynamic scaling"`.

### CODE REVIEW

**Comment by jerryzh168** [comment]
thanks and welcome Daniel, I added you to our contributor list so you don't need approval to run CI

**Review by vkuzo — APPROVED** [review]
thanks! Good to go after CI is green

**Pull request merged**
